lib/db_tests.js

```diff
@@ -179,4 +205,106 @@ describe('Database API', function() {
     });
   });
 
+  /* Advanced listing - restrictions
+   {"eq": { "a": 1 }}       data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   {"ne": { "a": 1 }}       data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------";
+   {"lt": { "a": 2 }}       data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   {"le": { "a": 2 }}       data.count === 2 ? "SUCCESS!!!!!" : "--------FAILURE--------";
+   {"gt": { "a": 1 }}       data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   {"ge": { "a": 1 }}       data.count === 2 ? "SUCCESS!!!!!" : "--------FAILURE--------";
+   {"in": { "a": [1] }}     data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   {"sort": { "a": -1 }}    data.list[0].fields.a === 2 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   {"skip": 0, "limit": 1}  data.count === 1 ? "SUCCESS!!!!!" : "--------FAILURE--------"
+   */
+
+  it('advanced listing - Equals 1', function (done) {
```

Comment:
Test cases should be named as "should _do whatever_", so that they explicitly describe the functionality. This one could be something like `it should filter using equal-to condition` and anyone would instantly know what is being proved.